lib/checkother.cpp

```diff
@@ -4367,6 +4367,33 @@ void CheckOther::overlappingWriteFunction(const Token *tok, const std::string& f
     reportError(tok, Severity::error, "overlappingWriteFunction", "Overlapping read/write in " + funcname + "() is undefined behavior");
 }
 
+void CheckOther::checkSuspiciousComma()
+{
+    if (!mSettings->severity.isEnabled(Severity::style)) {
+        return;
+    }
+
+    logChecker("CheckOpComma::warnSuspiciousComma");
+
+    for (const Token* tok = mTokenizer->list.front(); tok; tok = tok->next()) {
+        if (tok->str() == "," && tok->isBinaryOp()) {
+            const Token * parent = tok->astParent();
+            if (parent && (Token::simpleMatch(parent->previous(), "if (") ||
+                           Token::simpleMatch(parent->previous(), "while ("))) {
+                if (tok->previous()->str() == ")" && tok->previous()->link()->str() == "(") {
```

Comment:
I like overall to be over-explicit.  but this ` && tok->previous()->link()->str() == "("` is too redundant please remove that.